aptos-move/framework/aptos-stdlib/sources/cryptography/algebra.move

```diff
@@ -42,22 +214,61 @@ module aptos_std::algebra {
         serialize_internal<S, F>(element.handle)
     }
 
-    //
-    // (Public functions end here.)
-    // Native functions begin.
-    //
+    /// Get the order of group `G`, a big integer little-endian encoded as a byte array.
+    public fun order<S>(): vector<u8> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        order_internal<S>()
+    }
 
-    native fun deserialize_internal<G, F>(bytes: &vector<u8>): (bool, u64);
-    native fun add_internal<F>(handle_1: u64, handle_2: u64): u64;
-    native fun serialize_internal<G, F>(handle: u64): vector<u8>;
+    /// Cast an element of a structure `S` to a parent structure `L`.
+    public fun upcast<S,L>(element: &Element<S>): Element<L> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        Element<L> {
+            handle: upcast_internal<S,L>(element.handle)
+        }
+    }
+
+    /// Try casting an element `x` of a structure `L` to a sub-structure `S`.
+    /// Return none if `x` is not a member of `S`.
+    ///
+    /// NOTE: Membership check is performed inside, which can be expensive, depending on the structures `L` and `S`.
+    public fun downcast<L,S>(element_x: &Element<L>): Option<Element<S>> {
+        abort_unless_cryptography_algebra_natives_enabled();
+        let (succ, new_handle) = downcast_internal<L,S>(element_x.handle);
+        if (succ) {
+            some(Element<S> { handle: new_handle })
+        } else {
+            none()
+        }
+    }
+
+    /// Hash an arbitrary-length byte array `msg` into structure `S` using the given `suite`.
```

Comment:
Some notational typos here. Rewrite as:

```
Hash an arbitrary-length byte array `msg` into structure `St` using the given hashing method `Su`.
```

Also, wouldn't it be clearer to use `S, H` instead of `St, Su`?

Reply:
Fixed.